hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieSnapshotExporter.java

```diff
@@ -0,0 +1,219 @@
+    String snapshotPrefix;
+
+    @Parameter(names = {"--output-format", "-of"}, description = "e.g. Hudi or Parquet", required = true)
+    String outputFormat;
+
+    @Parameter(names = {"--output-partition-field", "-opf"}, description = "A field to be used by Spark repartitioning")
+    String outputPartitionField;
+  }
+
+  public void export(SparkSession spark, Config cfg) throws IOException {
+    String sourceBasePath = cfg.basePath;
+    String targetBasePath = cfg.outputPath;
+    String snapshotPrefix = cfg.snapshotPrefix;
+    String outputFormat = cfg.outputFormat;
+    String outputPartitionField = cfg.outputPartitionField;
+    JavaSparkContext jsc = new JavaSparkContext(spark.sparkContext());
+    FileSystem fs = FSUtils.getFs(sourceBasePath, jsc.hadoopConfiguration());
+
+    final SerializableConfiguration serConf = new SerializableConfiguration(jsc.hadoopConfiguration());
+    final HoodieTableMetaClient tableMetadata = new HoodieTableMetaClient(fs.getConf(), sourceBasePath);
+    final TableFileSystemView.BaseFileOnlyView fsView = new HoodieTableFileSystemView(tableMetadata,
+        tableMetadata.getActiveTimeline().getCommitsTimeline().filterCompletedInstants());
+    // Get the latest commit
+    Option<HoodieInstant> latestCommit =
+        tableMetadata.getActiveTimeline().getCommitsTimeline().filterCompletedInstants().lastInstant();
+    if (!latestCommit.isPresent()) {
+      LOG.warn("No commits present. Nothing to snapshot");
+      return;
+    }
+    final String latestCommitTimestamp = latestCommit.get().getTimestamp();
+    LOG.info(String.format("Starting to snapshot latest version files which are also no-late-than %s.",
+        latestCommitTimestamp));
+
+    List<String> partitions = FSUtils.getAllPartitionPaths(fs, sourceBasePath, false);
+    if (partitions.size() > 0) {
+      List<String> dataFiles = new ArrayList<>();
+
+      if (!StringUtils.isNullOrEmpty(snapshotPrefix)) {
+        for (String partition : partitions) {
+          if (partition.contains(snapshotPrefix)) {
+            dataFiles.addAll(fsView.getLatestBaseFilesBeforeOrOn(partition, latestCommitTimestamp).map(f -> f.getPath()).collect(Collectors.toList()));
+          }
+        }
+      } else {
+        for (String partition : partitions) {
+          dataFiles.addAll(fsView.getLatestBaseFilesBeforeOrOn(partition, latestCommitTimestamp).map(f -> f.getPath()).collect(Collectors.toList()));
+        }
+      }
+
+      if (!outputFormat.equalsIgnoreCase("hudi")) {
+        // Do transformation
+        if (!StringUtils.isNullOrEmpty(outputPartitionField)) {
+          // A field to do simple Spark repartitioning
+          spark.read().parquet(JavaConversions.asScalaIterator(dataFiles.iterator()).toSeq())
+              .repartition(new Column(outputPartitionField))
+              .write()
```

Comment:
I think you'd also need to partition the output dir
```suggestion
              .write()
              .partitionBy(outputPartitionField)
```